### DISCUSSION TITLE
macOS - How to customize app delegate and handle menu bar events

### DISCUSSION THREAD

**Discussion by jaigak** [unoplatform/uno#6977] [Q&A] [answered]
How do I customize the app delegate in a Uno Platform app to handle menu bar events on macOS? Can I just create my own AppDelegate class?

**Comment by jaigak** [comment, marked as answer]
I was able to add the required methods in the shared app class and export it as an action however I needed to put it inside an #if statement so it won't be compiled for other platforms. This is fine however it also leads to code duplication.

**Reply by jeromelaban** [reply]
You implemented correctly. The `App` class is used as the macOS app delegate. If you want to customize menu bar events, you can use `MenuBar` control with [an explicit style](https://github.com/unoplatform/uno/blob/56d3ba98ee6cab055dc8d7c4941ce4494e20e946/src/Uno.UI/UI/Xaml/Controls/MenuBar/MenuBar.xaml#L43) like this `{StaticResource NativeMenuBar}`.
